cmake/Dependencies.cmake

```diff
@@ -1314,6 +1314,9 @@ if(USE_ROCM)
        list(APPEND HIP_HIPCC_FLAGS -fdebug-info-for-profiling)
     endif(CMAKE_BUILD_TYPE MATCHES Debug)
 
+    # needed for compat with newer versions of hip-clang that introduced C++20 mangling rules
+    list(APPEND HIP_HIPCC_FLAGS -fclang-abi-compat=17)
```

Comment:
As we discussed in the ticket, we might want to guard this with version control.

Reply:
I just tested this change in rocm/pytorch:rocm5.7_ubuntu22.04_py3.10_pytorch_2.0.1 .  Build passed.

Reply:
KK. This change is more like a workaround until GCC also defaults to the new mangling, which I have no idea when. For now it just works fine, but in the future it will need to check the CMake C++ compiler ID and version and append the compile flag accordingly.